File: src/main/scala/com/github/kmizu/yapp/runtime/ParseError.scala
package com.github.kmizu.yapp.runtime

import java.lang.{Integer => JInteger}
import java.util.{Formatter => JFormatter}

case class ParseError(location: Location, message: String) {
  def line: Int = location.getLine

  def column: Int = location.getColumn

  def getLocation: Location = this.location

  def getLine: Int = location.getLine

  def getColumn: Int = location.getColumn

  def getMessage: String = message

  def getErrorMessage: String = {
    val builder = new StringBuilder

    val formatter = new JFormatter()(builder)
    formatter.format("%d, %d: %s", new JInteger(line), new JInteger(column), message)
    formatter.flush

    new String(builder.toString())
  }
}

File: src/jp/gr/java_conf/mizu/yapp/runtime/ParseError.java
package jp.gr.java_conf.mizu.yapp.runtime;

import java.util.Formatter;

public class ParseError {
  private final Location location;
  private final String message;
  
  public ParseError(Location location, String message) {
    this.location = location;
    this.message = message;
  }
  
  public Location getLocation() {
    return location;
  }
  
  public int getLine() {
    return location.getLine();
  }
  
  public int getColumn() {
    return location.getColumn();
  }
  
  public String getMessage() {
    return message;
  }
  
  public String getErrorMessage() {
    StringBuilder builder = new StringBuilder();
    Formatter f = new Formatter(builder);
    f.format("%d, %d: %s", location.getLine(), location.getColumn(), message);
    f.flush();
    return new String(builder);
  }
}
